### ISSUE TITLE
Model Flattening generates no fzn and ozn file

### ISSUE THREAD

**Issue by JulianKu** [paolodragone/pymzn#29]
Hi,
when I try running any model, for instance the 'knapsack' model, I get the error " 'NoneType' object has no attribute 'endswith' " which appears for the mzn_file variable in the args function being called from the _solve function.
I could backtrack the problem to the mzn2fzn function where the process is run but during that no fzn or ozn file is generated. This leads to the fzn_file and ozn_file variables being assigned the None object which then is handed over to the _solve function and causing the problem.

Has anyone ever had the same problem or knows how to deal with it?
Best regards,
Julian

**Comment by paolodragone** [comment]
Hi Julian,
The fact that fzn_file and ozn_file are assigned None is quite strange.
Could you post a piece of code to reproduce your problem?
Also, what OS are you using? and what version of MiniZinc? and what solver?

**Comment by JulianKu** [comment]
Thanks for your fast answer!

I am using Windows 10 with the current Minizinc version 2.2.1. The problem appeared using the default Gecode solver (I did not test others though).
I just took the example code from your readme and added the commands to set the paths for mzn2fzn and solns2out. The models I used are also just the ones from your readme. All lying in the same directory.

```Python
import pymzn
pymzn.config.set('mzn2fzn', "C:\Program Files\MiniZinc_IDE\mzn2fzn.exe")
pymzn.config.set('solns2out', "C:\Program Files\MiniZinc_IDE\solns2out.exe")

s = pymzn.minizinc('knapsack01.mzn', 'knapsack01.dzn', data={'capacity': 20})
print(s)
```

**Comment by paolodragone** [comment]
Got it! So the problem is I haven't touched the code in a while, and pretty much never properly tested it on Windows. Also, MiniZinc recently when through some major changes, to which I haven't yet adapted PyMzn to. The latest MiniZinc version which I tested it on is the 2.17. Still Windows might be an issue, but you can try (tell me if you do and it works! :).
I am sorry for the inconvenience, I will try to adapt to the latest MiniZinc version as soon as possible!

**Comment by JulianKu** [comment]
Thank you for your response. I will test it on MiniZinc 2.17 then and tell you.

**Comment by JulianKu** [comment]
So, testing it with Minizinc 2.17 worked under Windows. Thanks again!

**Comment by paolodragone** [comment]
Great!
I'll keep the issue open as a reminder for myself to solve the problem with newer versions of MiniZinc.
Cheers!

**Comment by paolodragone** [comment]
Fixed in 0.18.1.
